tests/src/performance/Scenario/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,485 @@
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var options = JitBenchHarnessOptions.Parse(args);
+
+            s_temporaryDirectory = Path.Combine(options.IntermediateOutputDirectory, "JitBench");
+            if (Directory.Exists(s_temporaryDirectory))
+                Directory.Delete(s_temporaryDirectory, true);
+            Directory.CreateDirectory(s_temporaryDirectory);
+
+            s_jitBenchDevDirectory = Path.Combine(s_temporaryDirectory, "JitBench-dev");
+
+            using (var h = new XunitPerformanceHarness(args))
+            {
+                ProcessStartInfo startInfo = Setup();
+                h.RunScenario(startInfo, () => { PrintHeader("Running Benchmark Scenario"); }, PostIteration, PostProcessing, s_ScenarioConfiguration);
+            }
+        }
+
+        static Program()
+        {
+            s_ScenarioConfiguration = new ScenarioConfiguration(TimeSpan.FromMilliseconds(20000)) {
+                Iterations = 11
+            };
+
+            // Set variables we will need to store results.
+            s_iteration = 0;
+            s_startupTimes = new double[s_ScenarioConfiguration.Iterations];
+            s_requestTimes = new double[s_ScenarioConfiguration.Iterations];
+        }
+
+        private static void DownloadAndExtractJitBenchRepo()
+        {
+            using (var client = new HttpClient())
+            {
+                // Download the JitBench repository and extract it.
+                const string repoUrl = "https://github.com/aspnet/JitBench";
+                const string sha1Id = "a44cd96e4ff8d7c1b43fe3c96a597919bb1544bd";
+                var archiveName = $"{sha1Id}.zip";
+                var url = $"{repoUrl}/archive/{archiveName}";
+                var zipFile = Path.Combine(s_temporaryDirectory, archiveName);
+
+                using (FileStream tmpzip = File.Create(zipFile))
+                {
+                    using (Stream stream = client.GetStreamAsync(url).Result)
+                        stream.CopyTo(tmpzip);
+                    tmpzip.Flush();
+                }
+
+                // If the repo already exists, we delete it and extract it again.
+                if (Directory.Exists(s_jitBenchDevDirectory))
+                    Directory.Delete(s_jitBenchDevDirectory, true);
+
+                // This step will create s_JitBenchDevDirectory.
+                ZipFile.ExtractToDirectory(zipFile, s_temporaryDirectory);
+                Directory.Move(Path.Combine(s_temporaryDirectory, $"JitBench-{sha1Id}"), s_jitBenchDevDirectory);
+            }
+        }
+
+        private static IDictionary<string, string> InstallSharedRuntime()
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = @".\Dotnet-Install.ps1 -SharedRuntime -InstallDir .dotnet -Channel master -Architecture x64"
```

Comment:
Can we start with Windows first? This is a port of an old PR, and it would be nice to start getting numbers through automation :)